utility/utils.py

```diff
@@ -146,6 +146,48 @@ def fuse_mount(fuse_clients, mounting_dir):
         log.error(e)
 
 
+def test_sync_via_bucket_stats(primary_rgw_node, secondary_rgw_node):
+    """
+    verify and monitor sync consistency via bucket stats across sites.
+    """
+    bucket_stat_pri_doc = json.loads(
+        primary_rgw_node.exec_command(cmd="sudo radosgw-admin bucket stats")[0]
+    )
+    bucket_stat_sec_doc = json.loads(
+        secondary_rgw_node.exec_command(cmd="sudo radosgw-admin bucket stats")[0]
+    )
+    total_buckets_pri = json.loads(
+        primary_rgw_node.exec_command(cmd="sudo radosgw-admin bucket list | wc -l")[0]
+    )
+    total_buckets_sec = json.loads(
+        secondary_rgw_node.exec_command(cmd="sudo radosgw-admin bucket list | wc -l")[0]
+    )
+    if total_buckets_pri == total_buckets_sec:
+        log.info("Number of buckets same on both sites, test data consistency now")
+        for bucket in range(0, total_buckets_pri - 2):
+            primary_objects = bucket_stat_pri_doc[bucket]["usage"]["rgw.main"][
```

Comment:
@viduship , could you make these lines more flexible like below to avoid errors if a bucket has no objects
bucket_stat_pri_doc[bucket]["usage"].get("rgw.main",{}).get("num_objects",0)

Reply:
Addressed @hmaheswa